planetiler-core/src/main/java/com/onthegomap/planetiler/util/CommonStringEncoder.java

```diff
@@ -51,4 +68,21 @@ public byte encode(String string) {
     }
     return result;
   }
+
+  /**
+   * Returns a int value to each unique string passed in.
+   *
+   * @param string the string to store
+   * @return an int that can be converted back to a string by {@link #decodeInt(int)}.
+   * @throws IllegalArgumentException if called for too many values
+   */
+  public int encodeInt(String string) {
+    // optimization to avoid more expensive computeIfAbsent call for the majority case when concurrent hash map already
+    // contains the value.
+    return stringToIdMap.computeIfAbsent(string, s -> {
```

Comment:
make sure you don't remove the `stringToIdMap.get(string) == null` optimization here - this is called billions of times for the planet, ConcurrentHashMap is optimized for reads, and all but 100-100k are read-only.